Completely replace SMS sending code.

meshy opened the pull request:
...and tidy up some.

meshy committed:
Tidy-up.

meshy committed:
Play with field requirements.

meshy committed:
Remove debug prints.

meshy committed:
Copy changes.

meshy committed:
Completely replace SMS sending code.

meshy commented:
@incuna/backend better?

meshy commented:
Any objections to making this public and adding it to pypi?

maxpeterson commented:
@meshy since it is from https://bitbucket.org/incuna/incuna-txtlocal which is already public I see little point in making your changes private.

ghickman commented:
![](http://24.media.tumblr.com/1f05d69e7e42024cf565c0d5f5a8a1db/tumblr_mgd3k01Re21qax99wo2_500.jpg)

maxpeterson commented on txtlocal/utils.py:
```diff
@@ -1,156 +1,68 @@
+    Any unrecognised kwargs will be passed to txtlocal in the POST data.
     """
+    post_data = {
+        'selectednums': '.'.join(recipient_list),
```
Is this correct?

meshy replied:
nope :)

ghickman commented:
@meshy – The changes you've made look ok at a cursory glance. Could you fix the `setup.py` while you're poking around?

meshy committed:
Fix typo.

jturnbull commented:
Needs a LICENCE and a CHANGELOG

meshy committed:
Fix up setup.py

meshy commented:
Ok, will also add a README.

meshy commented:
fwiw... also needs tests :(

ghickman commented on setup.py:
```diff
@@ -1,14 +1,13 @@
-    author = "Incuna Ltd",
-    author_email = "admin@incuna.com",
-    url = "http://incuna.com/",
+    install_requires=['requests'],
```
This should have a `>=` on it.

meshy replied:
Any suggestions for required version?

ghickman replied:
Whatever you used to develop it =)

meshy committed:
Semi-pin version of requirement.

meshy committed:
Add TXTLOCAL_ENDPOINT setting.

meshy committed:
Add readme

meshy committed:
Add LICENSE

meshy committed:
Do versioning correctly, and add CHANGELOG.

ghickman committed:
Fix import

ghickman committed:
Fix typo

ghickman committed:
Convert readme to rst

ghickman committed:
Use readme in long description

ghickman commented:
Couple of pedantic tweaks ready for rolling a release.

@maxpeterson – Are you happy to merge this now?

maxpeterson commented:
@ghickman it looks OK but has it actually been tested - as in sending an SMS?

ghickman committed:
Fix settings call

ghickman committed:
Rename post data variable to be more inline with requests default

ghickman committed:
Don't urlencode the message

The urlencode function takes dict like objects and makes HTTP query
strings from them.

ghickman committed:
Bump version to 1.0

The changes added since 0.1 have rewritten the majoirty of the code base
and should be considered backwards incompatible.

ghickman commented:
@meshy – Interesting, that makes no sense to me… as you URI encode key values!
